icu4j/main/classes/core/src/com/ibm/icu/util/CodePointMap.java

```diff
@@ -0,0 +1,460 @@
+         */
+        public int getStart() { return start; }
+        /**
+         * @return the (inclusive) end code point
+         * @draft ICU 63
+         * @provisional This API might change or be removed in a future release.
+         */
+        public int getEnd() { return end; }
+        /**
+         * @return the range value
+         * @draft ICU 63
+         * @provisional This API might change or be removed in a future release.
+         */
+        public int getValue() { return value; }
+        /**
+         * Sets the range. When using {@link #iterator()},
+         * iteration will resume after the newly set end.
+         *
+         * @param start new start code point
+         * @param end new end code point
+         * @param value new value
+         * @draft ICU 63
+         * @provisional This API might change or be removed in a future release.
+         */
+        public void set(int start, int end, int value) {
+            this.start = start;
+            this.end = end;
+            this.value = value;
+        }
+    }
+
+    private final class RangeIterator implements Iterator<Range> {
+        private Range range = new Range();
+
+        @Override
+        public boolean hasNext() {
+            return -1 <= range.end && range.end < 0x10ffff;
+        }
+
+        @Override
+        public Range next() {
+            if (getRange(range.end + 1, null, range)) {
+                return range;
+            } else {
+                throw new NoSuchElementException();
+            }
+        }
+
+        @Override
+        public final void remove() {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    /**
+     * Iterates over code points of a string and fetches trie values.
+     * This does not implement java.util.Iterator.
+     *
+     * <pre>
+     * void onString(CodePointMap map, CharSequence s, int start) {
+     *     CodePointMap.StringIterator iter = map.stringIterator(s, start);
+     *     while (iter.next()) {
+     *         int end = iter.getIndex();  // code point from between start and end
+     *         useValue(s, start, end, iter.getCodePoint(), iter.getValue());
+     *         start = end;
+     *     }
+     * }
+     * </pre>
+     *
+     * <p>This class is not intended for public subclassing.
+     *
+     * @draft ICU 63
+     * @provisional This API might change or be removed in a future release.
+     */
+    public class StringIterator {
+        /**
+         * @internal
+         * @deprecated This API is ICU internal only.
+         */
+        @Deprecated
+        protected CharSequence s;
+        /**
+         * @internal
+         * @deprecated This API is ICU internal only.
+         */
+        @Deprecated
+        protected int sIndex;
+        /**
+         * @internal
+         * @deprecated This API is ICU internal only.
+         */
+        @Deprecated
+        protected int c;
```

Comment:
Code would be clearer with more meaningful variable names than "int c;"

Reply:
It is very common across ICU to use short names like c, ch, cp. I would prefer not to replace c with cp or codePoint now.